From 83ef84ab78b4bb99506c7c8a26616115d7b4e5a7 Mon Sep 17 00:00:00 2001
From: Dorran Howell <dorran.howell@gmail.com>
Date: Thu, 12 Jun 2025 16:55:58 +0200
Subject: [PATCH] chore: Bump requests to newer patch version (#20494)

* chore: Bump requests to newer patch version

Updated the pinned version of requests in datadog_checks_base to a
version without known CVEs.

* Run ddev dep freeze

Updated agent requirements based on the new checks_base dependencies.

* Add changelog entry

* Update changelog entry name

* Update datadog_checks_base/changelog.d/20494.security

---------

Co-authored-by: Kyle Neale <kyle.a.neale@gmail.com>
---
 agent_requirements.in                          | 2 +-
 datadog_checks_base/changelog.d/20494.security | 1 +
 datadog_checks_base/pyproject.toml             | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)
 create mode 100644 datadog_checks_base/changelog.d/20494.security

diff --git a/agent_requirements.in b/agent_requirements.in
index 8e73f21f11732..d0df18fbb9ad7 100644
--- a/agent_requirements.in
+++ b/agent_requirements.in
@@ -58,7 +58,7 @@ requests-ntlm==1.3.0
 requests-oauthlib==2.0.0
 requests-toolbelt==1.0.0
 requests-unixsocket2==0.4.2
-requests==2.32.3
+requests==2.32.4
 rethinkdb==2.4.10.post1
 securesystemslib[crypto,pynacl]==0.28.0
 semver==3.0.4
diff --git a/datadog_checks_base/changelog.d/20494.security b/datadog_checks_base/changelog.d/20494.security
new file mode 100644
index 0000000000000..262d47bf50376
--- /dev/null
+++ b/datadog_checks_base/changelog.d/20494.security
@@ -0,0 +1 @@
+Updates `requests` to 2.32.4
diff --git a/datadog_checks_base/pyproject.toml b/datadog_checks_base/pyproject.toml
index 7d9df3ea9549b..7746c0c8b8c11 100644
--- a/datadog_checks_base/pyproject.toml
+++ b/datadog_checks_base/pyproject.toml
@@ -46,7 +46,7 @@ "pywin32==308; sys_platform == 'win32'",
     "pyyaml==6.0.2",
     "requests-toolbelt==1.0.0",
     "requests-unixsocket2==0.4.2",
-    "requests==2.32.3",
+    "requests==2.32.4",
     "simplejson==3.19.3",
     "uptime==3.0.1",
     "wrapt==1.17.2",
